### ISSUE TITLE
netty SSL is wrong

### ISSUE THREAD

**Issue by linking12** [grpc/grpc-java#3593]
Please answer these questions before submitting your issue.

### What version of gRPC are you using?
grpc 1.7.0

### What JVM are you using (`java -version`)?
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)

### What did you do?
```
               <dependency>
			<groupId>io.netty</groupId>
			<artifactId>netty-tcnative</artifactId>
			<version>${netty-tcnative.version}</version>
			<classifier>${os.detected.classifier}</classifier>
		</dependency>
```
I  think do not need jetty alpn.

but:
```
Caused by: java.lang.ClassNotFoundException: org/eclipse/jetty/alpn/ALPN
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at io.grpc.netty.JettyTlsUtil.isJettyAlpnConfigured(JettyTlsUtil.java:34)
```

netty has not load native library 

In netty OpenSSL

 ```
  private static void loadTcNative() throws Exception {
        String os = PlatformDependent.normalizedOs();
        String arch = PlatformDependent.normalizedArch();

        Set<String> libNames = new LinkedHashSet<String>(4);
        // First, try loading the platform-specific library. Platform-specific
        // libraries will be available if using a tcnative uber jar.
        libNames.add("netty_tcnative_" + os + '_' + arch);
        if ("linux".equalsIgnoreCase(os)) {
            // Fedora SSL lib so naming (libssl.so.10 vs libssl.so.1.0.0)..
            libNames.add("netty_tcnative_" + os + '_' + arch + "_fedora");
        }
        // finally the default library.
        libNames.add("netty_tcnative");

        NativeLibraryLoader.loadFirstAvailable(SSL.class.getClassLoader(),
            libNames.toArray(new String[libNames.size()]));
    }
```
   //mybe is  
     <font face="#FF7F50">libNames.add("netty-tcnative-" + os + '-' + arch);</font>
  (https://repo1.maven.org/maven2/io/netty/netty-tcnative/2.0.6.Final/netty-tcnative-2.0.6.Final-osx-x86_64.jar)

the library name is:

netty-tcnative-2.0.6.Final-osx-x86_64.jar

**Comment by linking12** [comment]
<dependencies>
    <dependency>
      <groupId>io.netty</groupId>
      <artifactId>netty-tcnative-boringssl-static</artifactId>
      <version>2.0.5.Final</version>
    </dependency>
  </dependencies>

2.0.5.Final is not match netty 4.1.16.Final, Please change the doc about https://github.com/grpc/grpc-java/blob/master/SECURITY.md

https://github.com/netty/netty/issues/7319

**Comment by ejona86** [comment]
Fixed in #3611
